lib/storage/src/content_manager/shard_distribution.rs

```diff
@@ -46,20 +45,11 @@ impl ShardDistributionProposal {
         for &peer in known_peers {
             let shard_count_on_peer = known_shards
                 .iter()
-                .filter(|shard| match shard {
-                    Shard::Remote(remote_shard) if remote_shard.peer_id == peer => true,
-                    Shard::Local(_) if this_peer_id == peer => true,
-                    _ => false,
-                })
+                .filter(|(_shard_id, peer_id)| *peer_id == peer)
                 .count();
             min_heap.push(Reverse(PeerShardCount::new(shard_count_on_peer, peer)))
         }
 
-        // no known peers with shards - add minimal entry for this peer so it gets all the shards
```

Comment:
What was the rational for removing this code, I remember adding it to handle a special case when no shards exist yet.

Reply:
I though, that it should not be required to know which shard is local to be able to calculate the distribution. So instead of extra check I included local shard into `known_shards` explicitly. It looks like it simplified the code